src/vma/dev/ring_simple.cpp

```diff
@@ -777,10 +777,17 @@ bool ring_simple::rx_process_buffer(mem_buf_desc_t* p_rx_wc_buf_desc, void* pv_f
 
 		if (likely((si != NULL) && si->flow_tag_enabled())) { 
 			// will process packets with set flow_tag_id and enabled for the socket
+			if (unlikely(p_eth_h->h_proto == htons(ETH_P_8021Q))) {
```

Comment:
I wouldn't use `unlikely` here. If there's a vlan defined - then all the packets will have the vlan interface and the condition becomes `likely`.
In principal, `likely`/`unlikely` should be considered at the code flow level (like when looping - it's `unlikely` that the current iteration is the last one, etc...) and not for user level decisions (such as whether the user is "likely" or not to set a vlan).